scipy/special/_gufuncs.cpp

```diff
@@ -53,9 +53,15 @@ extern const char *sph_harm_all_doc;
 extern "C" int wrap_PyUFunc_getfperr() { return PyUFunc_getfperr(); }
 
 static PyModuleDef _gufuncs_def = {
-    .m_base = PyModuleDef_HEAD_INIT,
-    .m_name = "_gufuncs",
-    .m_size = -1,
+    PyModuleDef_HEAD_INIT,
+    "_gufuncs",
```

Comment:
I had to resolve 1-2 merge conflicts when backporting to gh-20933 but full testsuite was happy locally at least. I'll flush the full CI later today probably.